101-lab/lab-notes.md

```diff
@@ -0,0 +1,130 @@
+    - Get volunteer admins (at least two)
+    - Walk through the process of adding people to the namespace
+        - Mention the need to include `@github` and the fact that names are lowercase only
+    - Get the admins to work through the rest of the list
+- Tips and Tricks
+    - show off `oc explain` and `-h` flag
+    - mention the deployment vs deploymentconfig issue (there is a step where you're asked to issue an `oc rollout` command which won't work - you can skip this step)
+- Questions?
+
+## Mid-Lab Check-In
+
+- Questions? 
+- Talk about Infrastructure As Code
+    - TODO: Maybe let's build an example set of yaml files that produces everything from the lab? Then the teams can look at this and use it as an example!
+    - Why is Infrastructure as Code a good idea?
+    - How are yaml manifests laid out?
+    - How do I generate yaml files? How do I apply them?
+    - What about templates?
+
+## Wrap-Up
+
+- How far did everyone get?
+    - Should we leave the lab stuff up for you guys?
+- Questions?
+- Services and Best Practices (don't worry, there are links and notes that will be posted about this!)
+    - Databases
+        - Already used MongoDB - the second most popular DB on the cluster
+        - Postgres is the most popular, and can be implemented in a few ways:
+            - Patroni is most common; you can use our images in either `bcgov` or Artifactory!
+            - EDB Operator is for teams with a license - this nets you additional enterprise-level support. Talk to Olena!
+            - CrunchyDB offers both enterprise and community licenses. Still new to the cluster!
+        - Our community has built an awesome backup-container, super easy DB backups! Take advantage of it!
+    - Images and Artifacts
+        - Artifactory:
+            - an artifact repository that can be used to host/access docker images, libraries and packages, helm charts, etc. 
+            - includes artifact scanning so you can make sure to use secure artifacts and images!
+        - Common images: pre-built docker images we have created for teams to use!
+            - Includes Postgres, MongoDB, AppAssessment
+    - KeyCloak
+        - an SSO solution that lets your app use IDIR login, plus also GitHub IDs and BCeID
+    - Sysdig
+        - monitoring app so you can track your resource usage
+        - notifies you of problems, like if your pod goes down or if your PVC is getting full
+    - Vault
+        - encrypted secret storage!
+    - Network Policies
+        - All namespaces on our cluster are deny access to all network traffic by default. You must explicitly allow your pods to talk to each other.
+        - You didn't have to worry about this in the lab because we made a network policy for you, but you'll need to make one on your own.
+        - Make sure you allow only the network traffic you actually need (use RC as an example?)
```

Comment:
since RC will be in gold one day, let's make sure to either mention gold cluster, or use a silver resident like registry